IntlExtension.php

```diff
@@ -307,7 +314,7 @@ public function formatTime(Environment $env, $date, ?string $timeFormat = 'mediu
         return $this->formatDateTime($env, $date, 'none', $timeFormat, $pattern, $timezone, $calendar, $locale);
     }
 
-    private function createDateFormatter(?string $locale, ?string $dateFormat, ?string $timeFormat, string $pattern, \DateTimeZone $timezone, string $calendar): \IntlDateFormatter
+    private function createDateFormatter(?string $locale, ?string $dateFormat, ?string $timeFormat, string $pattern, ?\DateTimeZone $timezone, string $calendar): \IntlDateFormatter
```

Comment:
this change breaks the computation of `$hash` without a `dateFormatterPrototype` as it calls `getName()` on `null`

Reply:
You're right, I overlooked that.

`if ($this->dateFormatterPrototype)`, then that case is already handled.
How about calculating the hash with `'...' . ($timezone ?: '') . '...'` to handle the last case?

Reply:
well, the issue is also that you won't pass a `$timezone` to the `IntlDateFormatter` constructor right now, which changes the behavior compared to today (and may break things for the vast majority of users not using the prototype, in an effort to make the prototype respect its timezone)

Reply:
I force-pushed a fix for this